ydb/library/yql/dq/actors/input_transforms/dq_input_transform_lookup.cpp

```diff
@@ -0,0 +1,251 @@
+#include "dq_input_transform_lookup.h"
+#include <ydb/library/yql/minikql/mkql_string_util.h>
+#include <ydb/library/yql/minikql/mkql_node_serialization.h>
+#include <ydb/library/yql/minikql/mkql_type_builder.h>
+#include <ydb/library/actors/core/actor_bootstrapped.h>
+#include <ydb/library/yql/dq/actors/compute/dq_compute_actor_async_io_factory.h>
+#include <ydb/library/yql/minikql/mkql_node_builder.h>
+#include <ydb/library/yql/minikql/mkql_string_util.h>
+#include <ydb/library/actors/core/actor.h>
+#include <ydb/library/actors/core/hfunc.h>
+
+namespace NYql::NDq {
+
+namespace {
+ 
+enum class EOutputRowItemSource{None, InputKey, InputOther, LookupKey, LookupOther};
+using TOutputRowColumnOrder = std::vector<std::pair<EOutputRowItemSource, ui64>>; //i -> {source, indexInSource}
+
+class TInputTransformStreamLookup
+        : public NActors::TActorBootstrapped<TInputTransformStreamLookup>
+        , public NYql::NDq::IDqComputeActorAsyncInput
+{
+public:
+    TInputTransformStreamLookup(
+        std::shared_ptr<NKikimr::NMiniKQL::TScopedAlloc> alloc,
+        const NMiniKQL::THolderFactory& holderFactory,
+        ui64 inputIndex,
+        NUdf::TUnboxedValue inputFlow,
+        NActors::TActorId computeActorId,
+        NDqProto::TDqInputTransformLookupSettings&& settings,
+        TVector<size_t>&& inputJoinColumns,
+        TVector<size_t>&& lookupJoinColumns,
+        const TOutputRowColumnOrder& outputRowColumnOrder
+    )
+        : Alloc(alloc)
+        , HolderFactory(holderFactory)
+        , InputIndex(inputIndex)
+        , InputFlow(std::move(inputFlow))
+        , ComputeActorId(std::move(computeActorId))
+        , Settings(std::move(settings))
+        , InputJoinColumns(std::move(inputJoinColumns))
+        , LookupJoinColumns(std::move(lookupJoinColumns))
+        , OutputRowColumnOrder(outputRowColumnOrder)
+        , InputFlowFetchStatus(NUdf::EFetchStatus::Yield)
+    {
+        Y_ABORT_UNLESS(Alloc);
+    }
+
+    void Bootstrap() {
+        //TODO implement me
+    }
+
+private:
+private: //IDqComputeActorAsyncInput
```

Comment:
Do we need this comment?

Reply:
It aims to group methods that implement this interface